polyply/src/gen_coords.py

```diff
@@ -61,13 +61,26 @@ def find_starting_node_from_spec(topology, start_nodes):
             mol_idx = res_spec['mol_idx']
             node = list(_find_nodes(topology.molecules[mol_idx], res_spec))[0]
             start_dict[mol_idx] = node
+            topology.molecules[mol_idx].root = node
         else:
             for idx, molecule in enumerate(topology.molecules):
                 if molecule.mol_name == res_spec['molname']:
                     node = list(_find_nodes(molecule, res_spec))[0]
                     start_dict[idx] = node
     return start_dict
 
+def _initialize_cylces(topology, cycles, tolerance):
+    for mol_name in cycles:
+        for mol_idx in topology.mol_idx_by_name[mol_name]:
+            # initalize as dfs tree
+            molecule = topology.molecules[mol_idx]
+            if set(dict(nx.degree(molecule)).values()) != {1, 2}:
+                raise IOError("Only linear cyclic molecules are allowed at the moment.")
```

Comment:
Here us an example:
```
g = nx.Graph()
g.add_edges_from([(0,1), (1, 2), (2, 3), (3, 4), (4, 5), (3, 6)])
nx.bfs_tree(g, source=6)
```
Given the graph g, if we construct the bfs tree starting at 6, we will traverse the graph in order 6, 3,2,1,4,5,6. If we have a distance restraint 0, 5 that would not work, because node 3 has as predecessor only node 6. So the algorithm as implemented now just breaks. On a logic level that is ok, because position 3 would be placed before 0, but be dependent on the coordinates of 0 but NOT vice versa. 

So for arbitrary restraints one would have to order the graph traversal or the restraint order in such a way that each node that is a reference node is built before the target node. That can be done but is beyond the what this PR and I can do at the moment.

Reply:
Hmmn, ok. At least for the current PR this is fine then.

To help my understanding of the algorithm: do you need the shortest path between the two nodes? Or all paths? Then, along that path you set intermediate restraints for the construction?

For all paths, see https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.simple_paths.shortest_simple_paths.html#networkx.algorithms.simple_paths.shortest_simple_paths

Reply:
You need THE path between the nodes that is taken during the bfs traversal of the graph. And that path needs to run from reference to target node sequentially. If you check the SI document in the zip I sent, in section 3.2 or so there should be an explanation of the algorithm.

Reply:
I made a new issue about it. Perhaps we can use it to discuss this a litle more.